packages/core/src/reports/utils.ts

```
@@ -498,3 +330,1587 @@ export const getBusinessPortalCount = async (
+  return format
+}
+
+export const formatData = (data, filter) => {
+
+  const { dateRange, startDate, endDate, frequencyType } = filter
+
+  const formattedData = [...data]
+
+  formattedData.forEach(item => {
+
+    if (item.hasOwnProperty('frequency')) {
```

Biome

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)